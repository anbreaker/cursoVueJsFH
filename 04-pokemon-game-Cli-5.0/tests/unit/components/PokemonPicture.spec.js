import { shallowMount } from '@vue/test-utils'

import PokemonPicture from '@/components/PokemonPicture'

describe('PokemonPicture Component', () => {
  test('Should match the snapshot', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Should show the hidden image and the pokemon 150', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false,
      },
    })

    const image1 = wrapper.find('img')

    expect(image1.exists()).toBeTruthy()
    // TODO - check image 2 is hidden

    expect(image1.classes('hidden-pokemon')).toBeTruthy()

    expect(image1.attributes('src')).toBe(
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/undefined.svg'
    )
  })

  test('Should show the pokemon if showPokemon is set to true', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: true,
      },
    })

    const image1 = wrapper.find('#img1')

    expect(image1.exists()).toBeTruthy()

    expect(image1.classes('hidden-pokemon')).toBe(true)
    expect(image1.classes('fade-in')).toBe(false)
  })
})
