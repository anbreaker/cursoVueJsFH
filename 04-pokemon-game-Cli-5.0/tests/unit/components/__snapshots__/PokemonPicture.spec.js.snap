// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture Component Should match the snapshot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="hidden-pokemon" id="img1" alt="pokemon"></div>`;
